main.py

```diff
@@ -336,9 +338,36 @@ def handle_errors(self, event):
         The event is sent only if the error is related to a request made
         by flow_manager.
         """
-        xid = event.content["message"].header.xid.value
-        error_type = event.content["message"].error_type
-        error_code = event.content["message"].code
+        message = event.content["message"]
+
+        connection = event.source
+        switch = connection.switch
+
+        xid = message.header.xid.value
+        error_type = message.error_type
+        error_code = message.code
+        error_data = message.data.pack()
+
+        # Get the packet responsible for the error
+        error_packet = connection.protocol.unpack(error_data)
+
+        if message.code == BadActionCode.OFPBAC_BAD_OUT_PORT:
+            if hasattr(error_packet, 'actions'):
+                for action in error_packet.actions:
+                    iface = switch.get_interface_by_port_no(action.port)
+
+                    # Set interface to drop packets forwarded to it
+                    if iface:
+                        iface.config = PortConfig.OFPPC_NO_FWD
+            else:
+                for instruction in error_packet.instructions
```

Comment:
```suggestion
                for instruction in error_packet.instructions:
```

Reply:
Thanks, I updated the code and add some comments too.